src/services/projects.js

```diff
@@ -260,4 +266,175 @@ module.exports = class ProjectsHelper {
+									return responses.failureResponse({
+										message: `${entityType.value} not added`,
+										statusCode: httpStatusCode.bad_request,
+										responseCode: 'CLIENT_ERROR',
+										error: utils.errorObject('body', entityType.value),
+									})
+								}
+							}
+						} else if (typeof projectEntityData === 'object') {
+							// If projectEntityData is an object, check if the item is present in allEntities
+							const entitiesPresent = allEntities.result.find(
+								(entity) => entity.value === projectEntityData.value
+							)
+							if (!entitiesPresent) {
+								return responses.failureResponse({
+									message: `${entityType.value} not added`,
+									statusCode: httpStatusCode.bad_request,
+									responseCode: 'CLIENT_ERROR',
+									error: utils.errorObject('body', entityType.value),
+								})
+							}
+						}
+					} else {
+						// If has_entities is false, check if the key in projectData has a non-empty value
+						if (!projectEntityData || projectEntityData === '') {
+							return responses.failureResponse({
+								message: `${entityType.value} should not be empty`,
+								statusCode: httpStatusCode.bad_request,
+								responseCode: 'CLIENT_ERROR',
+								error: utils.errorObject('body', entityType.value),
+							})
+						}
+						if (entityType.value !== 'tasks') {
+							const normalizedValue = unidecode(projectEntityData)
+							if (specialCharRegex.test(normalizedValue)) {
+								return responses.failureResponse({
+									message: `Special characters not allowed in ${entityType.value}`,
+									statusCode: httpStatusCode.bad_request,
+									responseCode: 'CLIENT_ERROR',
+									error: utils.errorObject('body', entityType.value),
+								})
+							}
+						}
+					}
+				}
+
+				if (projectData.tasks.length < 1) {
+					return responses.failureResponse({
+						message: 'TASK_NOT_FOUND',
+						statusCode: httpStatusCode.bad_request,
+						responseCode: 'CLIENT_ERROR',
+						error: utils.errorObject('body', 'tasks'),
+					})
+				}
+				for (let i = 0; i < projectData.tasks.length; i++) {
+					if (projectData.tasks[i].allow_evidences == common.TRUE) {
+						if (projectData.tasks[i].evidence_details.file_types.length < 1) {
+							return responses.failureResponse({
+								message: 'FILE_TYPE_NOT_SELECTED',
+								statusCode: httpStatusCode.bad_request,
+								responseCode: 'CLIENT_ERROR',
+								error: utils.errorObject('body', 'file_types'),
+							})
+						}
+					}
+					if (projectData.tasks[i].type === common.CONTENT) {
+						if (projectData.tasks[i].children.length < 1) {
+							return responses.failureResponse({
+								message: 'SUB_TASK_NOT_FOUND',
+								statusCode: httpStatusCode.bad_request,
+								responseCode: 'CLIENT_ERROR',
+								error: utils.errorObject('body', 'children'),
+							})
+						}
+					}
+				}
+				let updateProject = await resourceQueries.updateOne(
+					{ id: projectData.id },
```

Comment:
@ankit1shahu Are we validating only the project creator can update his project ? Add user id check also, Otherwise send error